staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go

```diff
@@ -417,15 +425,17 @@ func (cfgCtlr *configController) digestConfigObjects(newPLs []*flowcontrol.Prior
 
 		// if we are going to issue an update, be sure we track every name we update so we know if we update it too often.
 		currResult.updatedItems.Insert(fsu.flowSchema.Name)
-
-		enc, err := json.Marshal(fsu.condition)
+		conditions := objectForDigestConfigObjectsConditions{
+			Conditions: []flowcontrol.FlowSchemaCondition{fsu.condition},
+		}
+		status := objectForDigestConfigObjectsStatus{Status: conditions}
+		patchBytes, err := json.Marshal(status)
```

Comment:
we can reuse the existing type, right?
```
		status := flowcontrol.FlowSchemaStatus{
			Conditions: []flowcontrol.FlowSchemaCondition{
				fsu.condition,
			},
		}
		patchBytes, err := json.Marshal(status)
```

Reply:
`FlowSchemaStatus` has only one field today:
```
type FlowSchemaStatus struct {
	// `conditions` is a list of the current states of FlowSchema.
	// +listType=map
	// +listMapKey=type
	// +optional
	Conditions []FlowSchemaCondition `json:"conditions,omitempty" protobuf:"bytes,1,rep,name=conditions"`
}
```

looking into the future, if we add more fields inside `Status`, the patch bytes could include the default values for these new fields and that could cause an error. 
Can you please move this block to a new function
```
func generatePatchBytes(condition flowcontrol.FlowSchemaCondition) ([]byte, error) 
```
and then add a unit test that validates that the generated patch bytes are equal to expected, similar to what you have shown in the PR description `{"status":{"conditions":["1"]}}`

Reply:
The patch needs to include the `{"status":`, which will not come from marshaling a FlowSchemaStatus.

Reply:
I meant something like this that reuses the existing type:
```
func generatePatchBytes(condition flowcontrol.FlowSchemaCondition) ([]byte, error) {
	object := struct {
		Status flowcontrol.FlowSchemaStatus `json:"status"`
	} {
		Status: flowcontrol.FlowSchemaStatus{
			Conditions: []flowcontrol.FlowSchemaCondition{
				condition,
			},
		},
	}
	return json.Marshal(object)
}
```

Reply:
oh, good point, the outer type could be local to the one function needing it.

Reply:
Just changed the code this way. Thanks for the pointer super helpful.